docs/api/v3.rst

```diff
@@ -0,0 +1,487 @@
+
+
+.. TODO:
+
+   We could extend the scopes to be per-project or per-user. I'm not
+   sure if this is possible yet, but we should consider it.
+
+   * per-project: the user could go to the Project Admin tab and create a Token for this specific project
+   * per-user: the user could go to their own User settings and create token for all his projects
+
+
+Resources
+---------
+
+Projects
+~~~~~~~~
+
+Project list
+++++++++++++
+
+.. http:get::  /api/v3/project/
+
+    Retrieve a list of all the projects of the current logged in user.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl https://readthedocs.org/api/v3/project/
+
+    **Example response**:
+
+    .. sourcecode:: js
+
+        {
+            "count": 25,
+            "next": "/api/v3/project/limit=10&offset=10",
+            "previous": null,
+            "results": [PROJECTS]
+        }
+
+    :>json integer count: Total number of Projects.
+    :>json string next: URI for next set of Projects.
+    :>json string previous: URI for previous set of Projects.
+    :>json array results: Array of ``Project`` objects.
+
+.. TODO:
+
+   Add query string filters to narrow the query:
+     * privacy level
+     * language
+     * programming language
+     * repo url
+     * repo type
+     * version active
+     * version built
+     * all database?
+
+
+Project details
++++++++++++++++
+
+.. http:get::  /api/v3/project/(string:slug)/
+
+    Retrieve details of a single project.
+
+    .. sourcecode:: js
+
+        {
+            "name": "Pip",
+            "slug": "pip",
+            "description": "Pip Installs Packages.",
+            "language": "en",
+            "programming_language": "py",
+            "repo": "https://github.com/pypa/pip",
+            "repo_type": "git",
+            "default_version": "stable",
+            "default_branch": "master",
+            "documentation_type": "sphinx_htmldir",
+            "canonical_url": "http://pip.pypa.io/en/stable/",
+            "links": [
+                {
+                    "href": "/api/v3/project/pip/users/",
+                    "rel": "users",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/versions/",
+                    "rel": "versions",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/builds/",
+                    "rel": "builds",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/domains/",
+                    "rel": "domains",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/redirects/",
+                    "rel": "redirects",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/notifications/",
```

Comment:
All of these URLs can be expressed in the other way:

    /api/v3/notifications/pip/

but for some endpoints doesn't look great. Example,

   /api/v3/versions/pip/stable/

I'm not 100% on either, though. In this way it's clear that you will be returned a `Version` resource and `/api/v3/project/pip/versions/stable/` maybe it's not that clear.

Reply:
Another comment here is that we are using a similar approach of nested URLs for the URLs site: `https://readthedocs.org/dashboard/(project-slug)/version/(version-slug)/`

Reply:
I would try hard to not have multiple endpoints that return essentially the same thing. I think that point is not that all API nesting is bad but more that it can go too far.

I prefer `/api/v3/project/pip/notifications/` as an endpoint to `/api/v3/notifications/pip/` (I would prefer `/api/v3/notifications/?project__slug=pip` to the 2nd one although not the 1st). Since I think `/api/v3/notifications/` -- notifications for all projects -- is not particularly useful I think it's fine to only expose the endpoint that is specific to a project.